sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -39,7 +39,7 @@ internal class ReceiverManager
             _receiverOptions = new ServiceBusReceiverOptions
             {
                 ReceiveMode = ProcessorOptions.ReceiveMode,
-                PrefetchCount = ProcessorOptions.PrefetchCount,
```

Comment:
I really don't like this but while doing some spiking and fiddling around I discovered the options are never aligned anymore after the initial cloning. I wonder if the whole approach needs an overhaul. For example it would be possible to use the options as a base and then clone and reassign them on every modification. Then they would properly represent the adjusted values also for `MaxConcurrentCalls` when they are updated with `UpdateConcurrency`

Reply:
It shouldn't matter if the options are not aligned for the concurrency settings as those are consumed in the processor rather than the manager. With Prefetch it is different as that is consumed here since it gets passed to the receiver options. I think we can probably avoid needing to re-recreate the receiver for prefetch changes by copying what was done in the old library here - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Core/MessageReceiver.cs#L245. We could make the Prefetch property in ServiceBusReceiver internally settable so that it can be modified by the processor.

Reply:
Will wrap my head around it and report back